app/RouteHandlers/SofortNotificationHandler.php

```diff
@@ -16,16 +17,28 @@ class SofortNotificationHandler {
 
 	private $ffFactory;
 
+	/**
+	 * @var Request
+	 */
+	private $request;
+
 	public function __construct( FunFunFactory $ffFactory ) {
 		$this->ffFactory = $ffFactory;
 	}
 
 	public function handle( Request $request ): Response {
-		$useCase = $this->ffFactory->newHandleSofortPaymentNotificationUseCase( $request->query->get( 'updateToken' ) );
+		$this->request = $request;
```

Comment:
oh no! Wider scope and not needed mutation!

As this is (1) a class with only a single public method, (2) the only write happens at the start of that method and (3) the value is needed all over the place, I'm opting for this approach. Otherwise you end up with an extra parameter in all methods. (It is also possible to introduce a new class, but that seems premature.)